src/sources/postgresql_metrics.rs

```
@@ -0,0 +1,836 @@
+
+    // options
+    if let Some(options) = config.get_options() {
+        params.push(("options", options.to_string()));
+    }
+
+    // application_name
+    if let Some(name) = config.get_application_name() {
+        params.push(("application_name", name.to_string()));
+    }
+
+    // ssl_mode, ignore default value (SslMode::Prefer)
+    match config.get_ssl_mode() {
+        SslMode::Disable => params.push(("sslmode", "disable".to_string())),
+        SslMode::Prefer => {} // default, ignore
+        SslMode::Require => params.push(("sslmode", "require".to_string())),
+        _ => {} // non_exhaustive
+    };
+
+    // host
+    for host in config.get_hosts() {
+        match host {
+            Host::Tcp(host) => params.push(("host", host.to_string())),
+            #[cfg(unix)]
+            Host::Unix(path) => params.push(("host", path.to_string_lossy().to_string())),
+        }
+    }
+
+    // port
+    for port in config.get_ports() {
+        params.push(("port", port.to_string()));
+    }
+
+    // connect_timeout
+    if let Some(connect_timeout) = config.get_connect_timeout() {
+        params.push(("connect_timeout", connect_timeout.as_secs().to_string()));
+    }
+
+    // keepalives, ignore default value (true)
+    if !config.get_keepalives() {
+        params.push(("keepalives", "1".to_owned()));
+    }
+
+    // keepalives_idle, ignore default value (2 * 60 * 60)
+    let keepalives_idle = config.get_keepalives_idle().as_secs();
+    if keepalives_idle != 2 * 60 * 60 {
+        params.push(("keepalives_idle", keepalives_idle.to_string()));
+    }
+
+    // target_session_attrs, ignore default value (TargetSessionAttrs::Any)
+    match config.get_target_session_attrs() {
+        TargetSessionAttrs::Any => {} // default, ignore
+        TargetSessionAttrs::ReadWrite => {
+            params.push(("target_session_attrs", "read-write".to_owned()))
+        }
+        _ => {} // non_exhaustive
```

Maybe we should add `unknown` for these rather than silently ignoring added enum vales?

---

I'll add a warning message for `non_exhaustive` enums